app/controllers/membership_controller.rb

```diff
@@ -0,0 +1,39 @@
+class MembershipController < ApplicationController
+  before_action :require_login
+  before_action :init, only: [:destroy, :create, :inactivate, :reactivate]
+  before_action :require_leader, only: [:inactivate, :destroy, :reactivate]
+
+  def init
+    @group = Group.find(params[:group_id])
+    @own_membership = current_user.memberships.find { |m| m.group == @group }
+  end
+
+  def create
+    if !@group.users_can_apply || @own_membership
+      unauthorized_page
+    else
+      membership = Membership.create(grp_id: @group.id, usr_id: current_user.id)
+      Post.create(grp_member_id: membership.id, pttip_id: Membership::DEFAULT_POST_ID)
+      redirect_to :back
+    end
+  end
+
+  def destroy
+    Membership.delete(params[:id])
```

Comment:
This will issue a DELETE query to the database. It would not run any Rails callbacks or anything. That can be really surprising in the future. Do `Membership.find(params[:id]).destroy` to run callbacks.